embeddings/evaluator/text_classification_evaluator.py

```diff
@@ -1,21 +1,44 @@
-from typing import Any, Dict, List, Sequence, Union
+from typing import Any, Dict, List, Type, Union
 
 import torch
 from numpy import typing as nptyping
 
+from embeddings.evaluator.evaluation_results import Predictions, TextClassificationEvaluationResults
 from embeddings.evaluator.metrics_evaluator import MetricsEvaluator
 from embeddings.metric.hugging_face_metric import HuggingFaceMetric
 from embeddings.metric.metric import Metric
 
 
-class TextClassificationEvaluator(MetricsEvaluator):
-    @property
+class TextClassificationEvaluator(MetricsEvaluator[TextClassificationEvaluationResults]):
     def metrics(
         self,
-    ) -> Sequence[Metric[Union[List[Any], nptyping.NDArray[Any], torch.Tensor], Dict[Any, Any]]]:
-        return [
-            HuggingFaceMetric("accuracy"),
-            HuggingFaceMetric("f1", compute_kwargs={"average": "macro"}),
-            HuggingFaceMetric("recall", compute_kwargs={"average": "macro"}),
-            HuggingFaceMetric("precision", compute_kwargs={"average": "macro"}),
-        ]
+    ) -> Dict[str, Metric[Union[List[Any], nptyping.NDArray[Any], torch.Tensor], Dict[Any, Any]]]:
+        return {
+            "accuracy": HuggingFaceMetric("accuracy"),
+            "f1_macro": HuggingFaceMetric("f1", compute_kwargs={"average": "macro"}),
+            "f1_micro": HuggingFaceMetric("f1", compute_kwargs={"average": "micro"}),
+            "f1_weighted": HuggingFaceMetric("f1", compute_kwargs={"average": "weighted"}),
+            "recall_macro": HuggingFaceMetric("recall", compute_kwargs={"average": "macro"}),
+            "recall_micro": HuggingFaceMetric("recall", compute_kwargs={"average": "micro"}),
+            "recall_weighted": HuggingFaceMetric("recall", compute_kwargs={"average": "weighted"}),
+            "precision_macro": HuggingFaceMetric("precision", compute_kwargs={"average": "macro"}),
+            "precision_micro": HuggingFaceMetric("precision", compute_kwargs={"average": "micro"}),
+            "precision_weighted": HuggingFaceMetric(
+                "precision", compute_kwargs={"average": "weighted"}
+            ),
+        }
+
+    @property
+    def evaluation_results_cls(self) -> Type[TextClassificationEvaluationResults]:
+        return TextClassificationEvaluationResults
```

Comment:
same as with SeqLabeling, could just use the `TextClassificationEvaluationResults` in `evaluate`